eclair-core/src/main/scala/fr/acinq/eclair/wire/LightningMessageCodecs.scala

```diff
@@ -59,6 +59,17 @@ object LightningMessageCodecs {
       ("yourLastPerCommitmentSecret" | optional(bitsRemaining, privateKey)) ::
       ("myCurrentPerCommitmentPoint" | optional(bitsRemaining, publicKey))).as[ChannelReestablish]
 
+  // Legacy nodes are supposed to encode an upfront_shutdown_script only if both sides advertised support for option_upfront_shutdown_script.
```

Comment:
I think that a node may include an optional shutdown script even if its peer does not support it:

>   - if both nodes advertised the `option_upfront_shutdown_script` feature:
>    - MUST include either a valid `shutdown_scriptpubkey` as required by `shutdown` >`scriptpubkey`, or a zero-length `shutdown_scriptpubkey`.
 > - otherwise:
  >  - MAY include a`shutdown_scriptpubkey`.